### PR TITLE
Trilinos 12.2.1 (linuxbrew fix)

### PR DESCRIPTION
**Pull request by davydden**
Disable OpenMP for now as this leads to compiler errors with deal.ii

### CODE REVIEW

File: trilinos.rb [hunk 1]
```diff
@@ -123,7 +124,13 @@ def install
...
 
     args << onoff("-DTPL_ENABLE_MPI:BOOL=",         (build.with? "mpi"))
-    args << onoff("-DTrilinos_ENABLE_OpenMP:BOOL=", (ENV.compiler != :clang))
+    # TODO:
+    # OpenMP leads deal.II to fail with compiler errors in trilinos headers even though trilinos compiles fine
+    # It could be that there is a missing #include somewhere in Trilinos which becames visible when we
+    # try to use it.
+    # For now disable OpenMP:
+    # args << onoff("-DTrilinos_ENABLE_OpenMP:BOOL=", (ENV.compiler != :clang))
+    args << "-DTrilinos_ENABLE_OpenMP:BOOL=OFF"
```
Comment (dpo) [inline trilinos.rb]: Does this only happen on Linux? What about gcc on OSX?

Comment (davydden) [inline trilinos.rb]: likely to be the same issue. I don't use gcc on OSX, though.

**Comment by dpo** [comment]
There's a problem on line 227 because the `superlu` dep was commented out.

**Comment by davydden** [comment]
It has nothing to do with this PR o Superlu 5.0. There are compiler errors, i notifed developers on the mailing list about two weeks ago, no reply so far...

**Comment by dpo** [comment]
I remember. What I'm saying is that lines with `build.with? "superlu"` should be commented out as well.

**Comment by dpo** [comment]
There's a similar issue with other commented-out dependencies, e.g., `petsc`.

**Comment by davydden** [comment]
Unless you feel strong about it, i would keep it the it is now: when `superlu` is fixed, it's going to be one line to un-comment. Same applies to `petsc`. Everything else will be ready to use.

**Comment by dpo** [comment]
The cleanest would probably be to fix in a separate PR and then rebase this one.

**Comment by davydden** [comment]
With all due respect, i don't think it's a good idea. In addition to `petsc` and `superlu` mentioned above, one would also need to comment all the lines related to `lemon`, `cask`, `qd`, `binutils`, `csparse`. The current state folllows what we agreed in https://github.com/dpo/homebrew-trilinos/blob/master/new-trilinos.rb . Therefore, I will keep this PR as is.

**Comment by dpo** [comment]
I don't know why it wasn't brought up at the time. It just seems inconsistent and it's an easy fix. It's in #2759.

**Comment by dpo** [comment]
You can rebase this against the recent changes in `trilinos.rb`.

File: trilinos.rb (hunk 1 above)
Comment (dpo) [inline trilinos.rb]: Was this reported upstream?

Comment (davydden) [inline trilinos.rb]: i still investigate whether or not adding some #include can fix the issue. As soon as I will know more, i'll post to the mailing list

Comment (davydden) [inline trilinos.rb]: i reported the issue to the mailing list. Indeed it appears to be a Trilinos bug.

Comment (dpo) [inline trilinos.rb]: Thanks!

**Comment by davydden** [comment]
shall we merge this one?

**Comment by dpo** [comment]
There was an issue with the bottles, which is now resolved.

**Pull request closed without merging**